tests/unittests/sources/test_ec2.py

```diff
@@ -918,14 +918,14 @@ def test_ec2_local_returns_false_on_bsd(
     @pytest.mark.usefixtures("disable_netdev_info")
     @mock.patch("cloudinit.net.ephemeral.EphemeralIPv6Network")
     @mock.patch("cloudinit.net.ephemeral.EphemeralIPv4Network")
-    @mock.patch("cloudinit.distros.net.find_fallback_nic")
+    @mock.patch("cloudinit.distros.net.find_candidate_nics")
```

Comment:
Thanks for updating this unit test, but can we get a unit test that tests the new looping behavior?

Reply:
Added a new unit test: test_ec2_local_get_metadata_via_iterating_nics. Please take your time to review and let me know if it meets your expectations.